compiler/rustc_hir_analysis/src/astconv/bounds.rs

```diff
@@ -66,36 +214,41 @@ impl<'tcx> dyn AstConv<'tcx> + '_ {
             }
         }
 
-        if unbounds.len() > 1 {
-            tcx.dcx().emit_err(errors::MultipleRelaxedDefaultBounds {
-                spans: unbounds.iter().map(|ptr| ptr.span).collect(),
-            });
+        if unbounds.len() > 1 && !tcx.features().more_maybe_bounds {
+            self.tcx()
+                .sess
+                .create_feature_err(
+                    errors::MultipleRelaxedDefaultBounds {
+                        spans: unbounds.iter().map(|ptr| ptr.span).collect(),
+                    },
+                    sym::more_maybe_bounds,
+                )
+                .emit();
         }
 
-        let mut seen_sized_unbound = false;
+        let mut seen_unbound = false;
         for unbound in unbounds {
-            if let Some(sized_def_id) = sized_def_id
-                && unbound.trait_ref.path.res == Res::Def(DefKind::Trait, sized_def_id)
-            {
-                seen_sized_unbound = true;
-                continue;
+            let unbound_def_id = unbound.trait_ref.trait_def_id();
+            if unbound_def_id == Some(trait_def_id) {
+                seen_unbound = true;
             }
-            // There was a `?Trait` bound, but it was not `?Sized`; warn.
-            tcx.dcx().span_warn(
-                unbound.span,
-                "relaxing a default bound only does something for `?Sized`; \
-                all other traits are not bound by default",
-            );
-        }
 
-        if seen_sized_unbound || seen_negative_sized_bound || seen_positive_sized_bound {
-            // There was in fact a `?Sized`, `!Sized` or explicit `Sized` bound;
-            // we don't need to do anything.
-        } else if sized_def_id.is_some() {
-            // There was no `?Sized`, `!Sized` or explicit `Sized` bound;
-            // add `Sized` if it's available.
-            bounds.push_sized(tcx, self_ty, span);
+            let emit_relax_err = || {
+                tcx.dcx().span_warn(
+                    unbound.span,
+                    "relaxing a default bound only does something for `?Sized` and `experimental_default_bounds`; \
```

Comment:
This will be emitted on stable, please make sure `experimental_default_bounds` are only mentioned on nightly